internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -351,20 +405,117 @@ func resourceOrchestratedVirtualMachineScaleSet() *pluginsdk.Resource {
 				return false
 			}),
 
+			// SKU Profile validation
 			pluginsdk.CustomizeDiffShim(func(ctx context.Context, diff *pluginsdk.ResourceDiff, v interface{}) error {
 				skuName, hasSkuName := diff.GetOk("sku_name")
 				_, hasSkuProfile := diff.GetOk("sku_profile")
 
 				if hasSkuProfile {
 					if !hasSkuName || skuName != SkuNameMix {
-						return fmt.Errorf("`sku_profile` can only be configured when `sku_name` is set to `Mix`")
+						return fmt.Errorf("`sku_profile` can only be configured when `sku_name` is set to `Mix`, got `%s`", skuName)
 					}
 				} else {
 					if hasSkuName && skuName == SkuNameMix {
 						return fmt.Errorf("`sku_profile` must be configured when `sku_name` is set to `Mix`")
 					}
 				}
 
+				// Validate SKU Profile configuration
+				if skuProfileRaw := diff.Get("sku_profile").([]interface{}); len(skuProfileRaw) > 0 {
+					skuProfile := skuProfileRaw[0].(map[string]interface{})
+					allocationStrategy := skuProfile["allocation_strategy"].(string)
+
+					configRaw := diff.GetRawConfig().AsValueMap()["sku_profile"]
+					if configRaw.IsNull() || !configRaw.IsKnown() {
+						return nil
+					}
+
+					skuProfiles := configRaw.AsValueSlice()
+					if len(skuProfiles) == 0 {
+						return nil
+					}
+
+					// TODO: Remove in v5.0
+					if !features.FivePointOh() {
+						vmSizesProvided := len(skuProfile["vm_sizes"].(*pluginsdk.Set).List()) > 0
+						vmSizeProvided := len(skuProfile["vm_size"].(*pluginsdk.Set).List()) > 0
+						if !vmSizesProvided && !vmSizeProvided {
+							return fmt.Errorf("either `vm_sizes` or `vm_size` must be configured in `sku_profile`")
+						}
```

Comment:
could we use the schema to validate this?

Reply:
Maybe? I will try adding a `AtLeastOneOf` to the schema for both of these fileds/blocks in the `!features.FivePointOh()` schema and see if that works.

Reply:
I cannot enforce this at the schema level, due to the following limitation of the `AtLeastOneOf ` schema attribute:
```text
Only absolute attribute paths, ones starting with top level attribute names,
are supported. Attribute paths cannot be accurately declared for 
TypeList (if MaxItems is greater than 1), TypeMap, or TypeSet attributes.
To reference an attribute under a single configuration block (TypeList with
Elem of *Resource and MaxItems of 1), the syntax is "parent_block_name.0.child_attribute_name".
```